### PR TITLE
feat(encryption)

### PR DESCRIPTION
**Pull request by henrikstranneheim**
## Description
 Refactor encryption dir in config

### Changed

- Remove dict and legacy entry. Replace with string
- Decouple from file system path when querying PDC
- Add flow cell specific search pattern
- Simplify extracting file path from DCMS output

### How to prepare for test

- [ ] Ssh to relevant server (depending on type of change)
- [ ] Use stage: `us`
- [ ] Paxa the environment: `paxa`
- [ ] Install on stage (example for Hasta):
    ```shell
    bash /home/proj/production/servers/resources/hasta.scilifelab.se/update-tool-stage.sh -e S_main -t cg -b flat_encrypt_dir_2 -a
    ```

### How to test

- [ ] Do ...

### Expected test outcome

- [ ] Check that ...
- [ ] Take a screenshot and attach or copy/paste the output.

## Review

- [ ] Tests executed by
- [ ] "Merge and deploy" approved by
Thanks for filling in who performed the code review and the test!

### This [version](https://semver.org/) is a

- [ ] **MAJOR** - when you make incompatible API changes
- [ ] **MINOR** - when you add functionality in a backwards compatible manner
- [x] **PATCH** - when you make backwards compatible bug fixes or documentation/instructions

## Implementation Plan

- [ ] Document in ...
- [ ] Deploy this branch on ...
- [ ] Inform to ...

### CODE REVIEW

**Review by ChrOertlin — APPROVED** [review]
Looks good! Just a comment about a possibility of having the flow cell field `has_backup` to be updated here. Not sure if this would be the right place.

File: cg/cli/backup.py [hunk 1]
```diff
@@ -39,7 +39,7 @@ def fetch_flow_cell(context: CGConfig, dry_run: bool, flow_cell_id: Optional[str
     tar_api = TarAPI(binary_path=context.tar.binary_path, dry_run=dry_run)
     context.meta_apis["backup_api"] = BackupAPI(
         encryption_api=encryption_api,
-        encrypt_dir=context.backup.encrypt_dir.dict(),
+        encrypt_dir=context.backup.encrypt_dir,
```
[the remark this answers is not in the document]
Comment (henrikstranneheim) [reply cg/cli/backup.py:42]: We should do it when we archive, but we should add a check here before fetching, but not in this PR. Will do it in the coming ones 👍 .

**Comment by henrikstranneheim** [comment]
Redid tests on stage ✅ 
```
θ63° 5d [hiseq.clinical@hasta:~] [S_base] 24s $ cg backup  fetch-flow-cell -d -f HJFW5DSXX
Running command /bin/dsmc q archive */*HJFW5DSXX*.gpg
Found archived files for PDC query: */*HJFW5DSXX*.gpg
Encryption key found: /home/hiseq.clinical/ENCRYPT/190329_A00689_0018_AHJFW5DSXX.key.gpg
Flow cell found: /home/hiseq.clinical/ENCRYPT/190329_A00689_0018_AHJFW5DSXX.tar.gz.gpg
θ66° 5d [hiseq.clinical@hasta:~] [S_base] 26s $
```

**Comment by henrikstranneheim** [comment]
Deployed to production:
```
Log deploy... done.
cg, version 50.1.4
```

**Pull request merged**